extension/src/experiments/workspace.ts

```diff
@@ -178,43 +153,25 @@ export class WorkspaceExperiments extends BaseWorkspaceWebviews<
     commandId: ModifiedExperimentAndRunCommandId,
     overrideRoot?: string
   ) {
-    const cwd = await this.getDvcRoot(overrideRoot)
-    if (!cwd) {
+    const { cwd, repository } = await this.getRepositoryAndCwd(overrideRoot)
+    if (!(cwd && repository)) {
       return
     }
 
-    const repository = this.getRepository(cwd)
-    if (!repository) {
-      return
-    }
-
-    const shouldContinue = await this.checkOrAddPipeline(cwd)
-    if (!shouldContinue) {
-      return
-    }
-
-    return await repository.modifyWorkspaceParamsAndRun(commandId)
+    return await repository.modifyWorkspaceParamsAndRun(commandId, cwd)
```

Comment:
[F] We need to pass a cwd because the pipeline is no longer assumed to be at the root.